### PR TITLE
[Serve] Update AIR Examples to use new API, add linked guide

### LINKED ISSUE #27447: [serve][docs] `Serving Ray AIR Checkpoints` user guide

### PR DESCRIPTION
**Pull request by simon-mo**
## Why are these changes needed?
- Updated all examples mentioning Ray Serve in Ray AIR documentation to use the new API. I didn't re-run the notebook because that will mess up the output with Ray Core and pandas deprecation warnings. I will rely the CI infra to tell me the code is correct. 
- Added a new page in Ray Serve to simply linked to Ray AIR's serving guide. This encourages discovery from serve users to find out more about Ray AIR. edit suggestion welcomed!

## Related issue number
Closes #27447

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by sihanwang41 — APPROVED** [review]

File: doc/source/serve/ray-air-serve-guide.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Serving Ray AIR `Checkpoint`
```
Comment (edoakes) [inline doc/source/serve/ray-air-serve-guide.md:1]:
```suggestion
# Serving Ray AIR `Checkpoint`s
```

File: doc/source/serve/user-guide.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ you will learn
 - [Configuring HTTP and Bring Your own FastAPI app](http-guide)
 - [Using HTTP Adapters](http-guide)
 - [Composing Deployments](handle-guide)
+- [Serving Ray AIR Checkpoint](ray-air-serve-guide)
```
Comment (edoakes) [inline doc/source/serve/user-guide.md:13]:
```suggestion
- [Serving Ray AIR Checkpoints](ray-air-serve-guide)
```

**Review by edoakes — APPROVED** [review]

**Comment by edoakes** [comment]
please fix plural nit then LGMT

**Comment by edoakes** [comment]
@dmatrix please take a look when you have a chance. Pretty much just code changes.

File: doc/source/serve/ray-air-serve-guide.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Serving Ray AIR `Checkpoint`s
+
+[Ray AI Runtime (AIR)](air) is a scalable and unified toolkit for ML applications. Ray Serve is part of Ray AIR.
+It natively integrates with AIR's `Predictor` and `Checkpoint` abstraction to enable seamless transition from training to serving.
+You can learn more about how to serve Ray AIR Checkpoint in [this guide](air-serving-guide).
```
Comment (stephanie-wang) [inline doc/source/serve/ray-air-serve-guide.md:5]:
```suggestion
It natively integrates with AIR's `Predictor` and `Checkpoint` abstractions to enable seamless transition from training to serving.
You can learn more about how to serve Ray AIR Checkpoints in [this guide](air-serving-guide).
```

**Review by stephanie-wang — APPROVED** [review]

**Comment by edoakes** [comment]
Thanks @stephanie-wang!

@simon-mo, the DCO build is failing.

**Pull request merged**